packages-exp/auth-exp/src/core/providers/anonymous.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AuthProvider, ProviderId, SignInMethod } from '.';
+import { signUp } from '../../api/authentication/sign_up';
+import { Auth } from '../../model/auth';
+import { AuthCredential } from '../../model/auth_credential';
+import { IdTokenResponse } from '../../model/id_token';
+import { debugFail } from '../util/assert';
+
+export class AnonymousCredential implements AuthCredential {
+  providerId = ProviderId.ANONYMOUS;
+  signInMethod = SignInMethod.ANONYMOUS;
+
+  toJSON(): never {
+    debugFail('Method not implemented.');
```

Comment:
@samhorlbeck this is a fun way to handle unimplemented methods

Reply:
yeah that's clever